File: 3028-ant-on-the-boundary/3028-ant-on-the-boundary.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var returnToBoundaryCount = function(nums) {
    let total = 0;
    let boundaryHits = 0;

    nums.forEach(num => {
        total += num;
        if (total === 0) {
            boundaryHits += 1;
        }
    });

    return boundaryHits;
};

File: 3028-ant-on-the-boundary/3028-ant-on-the-boundary.rs

impl Solution {
    pub fn return_to_boundary_count(nums: Vec<i32>) -> i32 {
        let (mut total, mut boundary_hits) = (0, 0);

        for &num in nums.iter() {
            total += num;
            if total == 0 {
                boundary_hits += 1;
            }
        }

        boundary_hits
    }
}